src/EventStore.Client.Common/EventStoreClientSettingsExtensions.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Threading;
+using Grpc.Core;
+
+#nullable enable
+namespace EventStore.Client {
+	internal static class EventStoreCallOptions {
+		public static CallOptions Create(EventStoreClientSettings settings,
+			EventStoreClientOperationOptions operationOptions, UserCredentials? userCredentials,
+			CancellationToken cancellationToken) => new CallOptions(
+			cancellationToken: cancellationToken,
+			deadline: DeadlineAfter(operationOptions.TimeoutAfter),
+			headers: new Metadata(),
+			credentials: CallCredentials.FromInterceptor(async (context, metadata) => {
+				var credentials = settings.DefaultCredentials ?? userCredentials;
+
+				if (credentials == null) {
+					return;
+				}
+
+				var authorizationHeader = await settings.OperationOptions
+					.GetAuthenticationHeaderValue(credentials, CancellationToken.None)
+					.ConfigureAwait(false);
+				metadata.Add("Authorization", authorizationHeader);
```

Comment:
We have constants for the headers, we should use them or remove them.

```suggestion
				metadata.Add(Constants.Headers.Authorization, authorizationHeader);
```